hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MetricsRegionServerWrapperImpl.java

```diff
@@ -709,13 +710,50 @@ synchronized public void run() {
         long tempMobScanCellsSize = 0;
         long tempBlockedRequestsCount = 0;
         int regionCount = 0;
+
+        long tempReadRequestsCount = 0;
+        long tempWriteRequestsCount = 0;
+        long currentReadRequestsCount = 0;
+        long currentWriteRequestsCount = 0;
+        long lastReadRequestsCount = 0;
+        long lastWriteRequestsCount = 0;
+        long readRequestsDelta = 0;
+        long writeRequestsDelta = 0;
+        long totalReadRequestsDelta = 0;
+        long totalWriteRequestsDelta = 0;
+        String encodedRegionName;
+        ArrayList<String> hregions = new ArrayList<String>();
```

Comment:
Should be a HashSet to avoid paying a linear cost on `List.contains(Object)` down below.